src/modules/Users-Portal/Component/UsersShared/CalendarBooking/CalendarBooking.tsx

```diff
@@ -1,6 +1,202 @@
+import Grid from "@mui/material/Grid";
+import { useState } from "react";
+
+import img from "../../../../../assets/images/banner.png";
+import DatePicker from "../CalendarBooking/DatePicker/DatePicker";
+import { Box, Button, IconButton, TextField, Typography } from "@mui/material";
+import { Add, Remove } from "@mui/icons-material";
+import { axiosInstance, getRoomDetails } from "../../../../../services/urls";
+import dayjs from "dayjs";
+import { useNavigate } from "react-router-dom";
+import { AxiosError } from "axios";
+import { toast } from "react-toastify";
 
 export default function CalendarBooking() {
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const [dateRange, setDateRange] = useState<{
+    startDate: Date | null;
+    endDate: Date | null;
+  }>({
+    startDate: null,
+    endDate: null,
+  });
+  const [error, setError] = useState<string>("");
+
+  const navigate = useNavigate();
+  const [count, setCount] = useState(1);
+
+  const handlePopoverClose = () => {
+    setAnchorEl(null);
+  };
+  const handleIncrease = () => {
+    setCount(count + 1);
+  };
+
+  const handleDecrease = () => {
+    if (count > 1) {
+      setCount(count - 1);
+    }
+  };
+  const getRooms = async () => {
+    if (!dateRange.startDate || !dateRange.endDate) {
+      setError("Please pick a start and end date.");
+      return;
+    }
+
+    try {
+      const { startDate, endDate } = dateRange;
+
+      const response = await axiosInstance.get(getRoomDetails, {
+        params: {
+          startDate: dayjs(startDate).format("YYYY-MM-DD"),
+          endDate: dayjs(endDate).format("YYYY-MM-DD"),
+        },
+      });
+
+      console.log(dateRange);
+      console.log(response.data.data.rooms);
+
+      navigate("/explore", {
```

Comment:
to send the start date as search params, we can make this as
`/explore?startDate=${startDate}&endDate=${endDate}`

after formatting the dates